fundingmanager.go

```diff
@@ -1990,12 +1990,35 @@ func (f *fundingManager) waitForFundingConfirmation(
 		TxPosition:  uint16(fundingPoint.Index),
 	}
 
+	err = f.handleFundingConfirmation(completeChan, shortChanID)
+	if err != nil {
+		fndgLog.Errorf("Unable to handle funding confirmation: %v", err)
+		return
+	}
+
+	select {
+	case confChan <- &shortChanID:
+	case <-f.quit:
+		return
+	}
+}
+
+// handleFundingConfirmation marks a channel as open in the database, and set
+// the channelOpeningState markedOpen. In addition it will report the now
+// decided short channel ID to the switch, and close the local discovery signal
+// for this channel.
+func (f *fundingManager) handleFundingConfirmation(
```

Comment:
Why not also fold this into the `stateStep` method? This method has been deleted, then re-added (in a slightly different format), in this diff so far.

Reply:
It is a new method really, name colliding with the removed one is a unfortunate coincidence...

It cannot easily be folded into `stateStep`, since the state machine really starts from the `markedOpen` state, which is set here. Or maybe I misunderstood what you meant?